File: |-- src_R/| |-- 03_akar_kuadrat.R
#Akar real berbeda
# Data contoh
a <- 1
b <- -5
c <- 6

D <- b^2 - 4*a*c

if (D > 0) {
  
  x1 <- (-b + sqrt(D))/(2*a)
  x2 <- (-b - sqrt(D))/(2*a)
  
  cat(sprintf("Akar pertama = %.3f\n", x1))
  cat(sprintf("Akar kedua   = %.3f\n", x2))
  
} else if (D == 0) {
  
  x <- -b/(2*a)
  
  cat(sprintf("Akar kembar = %.3f\n", x))
  
} else {
  
  cat("Persamaan memiliki akar-akar imajiner\n")
  
}


#Akar kembar
# Nilai koefisien
a <- 1
b <- -4
c <- 4

# Menghitung diskriminan
D <- b^2 - 4*a*c

if (D == 0) {
  
  x <- -b/(2*a)
  
  cat(sprintf("Akar kembar = %.3f\n", x))
  
} else {
  
  cat("Bukan kasus akar kembar\n")
  
}


#Akar imajiner
# Nilai koefisien
a <- 1
b <- 2
c <- 5

# Menghitung diskriminan
D <- b^2 - 4*a*c

if (D < 0) {
  
  real <- -b/(2*a)
  imag <- sqrt(abs(D))/(2*a)
  
  cat(sprintf("x1 = %.3f + %.3fi\n", real, imag))
  cat(sprintf("x2 = %.3f - %.3fi\n", real, imag))
  
} else {
  
  cat("Bukan kasus akar imajiner\n")
  
}

File: |-- src_python/| |-- 03_akar_kuadrat.py
#Akar Real Berbeda
import math

def akar_kuadrat(a, b, c):

    D = b**2 - 4*a*c

    if D > 0:
        x1 = (-b + math.sqrt(D)) / (2*a)
        x2 = (-b - math.sqrt(D)) / (2*a)

        print(f"Akar pertama = {x1:.3f}")
        print(f"Akar kedua   = {x2:.3f}")

    elif D == 0:
        x = -b / (2*a)

        print(f"Akar kembar = {x:.3f}")

    else:
        print("Persamaan memiliki akar-akar imajiner")

a = float(input("Masukkan a : "))
b = float(input("Masukkan b : "))
c = float(input("Masukkan c : "))

akar_kuadrat(a,b,c)


#Akar kembar
import math

def akar_kuadrat(a, b, c):

    D = b**2 - 4*a*c

    if D > 0:
        x1 = (-b + math.sqrt(D)) / (2*a)
        x2 = (-b - math.sqrt(D)) / (2*a)

        print(f"Akar pertama = {x1:.3f}")
        print(f"Akar kedua   = {x2:.3f}")

    elif D == 0:
        x = -b / (2*a)

        print(f"Akar kembar = {x:.3f}")

    else:
        print("Persamaan memiliki akar-akar imajiner")

a = float(input("Masukkan a : "))
b = float(input("Masukkan b : "))
c = float(input("Masukkan c : "))

akar_kuadrat(a,b,c)


#Akar imajiner
import math

def akar_kuadrat(a, b, c):

    D = b**2 - 4*a*c

    if D > 0:
        x1 = (-b + math.sqrt(D)) / (2*a)
        x2 = (-b - math.sqrt(D)) / (2*a)

        print(f"Akar pertama = {x1:.3f}")
        print(f"Akar kedua   = {x2:.3f}")

    elif D == 0:
        x = -b / (2*a)

        print(f"Akar kembar = {x:.3f}")

    else:
        print("Persamaan memiliki akar-akar imajiner")

a = float(input("Masukkan a : "))
b = float(input("Masukkan b : "))
c = float(input("Masukkan c : "))

akar_kuadrat(a,b,c)
